be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -194,8 +202,19 @@ Status VerticalGeneralTabletWriter::flush_columns() {
     }
 
     DCHECK(_segment_writers[_current_writer_index]);
-    RETURN_IF_ERROR(flush_columns(&_segment_writers[_current_writer_index]));
+    RETURN_IF_ERROR(flush_columns(_segment_writers[_current_writer_index]));
     _current_writer_index = 0;
+
+    if (_segment_writer_finalize_token != nullptr) {
+        // Wait all async tasks finished
```

Comment:
Comment does not match the code, fix or remove the comment?

Reply:
Depending on the implemented code, it does not wait for all asynchronous tasks to finish when an error is encountered

Reply:
done